src/js/components/week.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+// import moment from 'moment';
+import helpers from './../util/helpers';
+import Day from './day.jsx';
+
+export default class Week extends React.Component {
+  render() {
+
+    return (
+      <div className = 'tsc-week'>
+        { this._renderWeekDays() }
+      </div>
+    );
+  }
+
+  _renderWeekDays() {
+    const {
+      weekToRender,
+    } = this.props;
+
+    return weekToRender.map((day, index) => {
+      let formattedDate = helpers.getMomentFromCalendarJSDateElement(day);
+      return (
+        <Day
+          key = { index }
+          onTimeslotClick = { this._onTimeslotClick.bind(this, index) }
+          momentTime = { formattedDate }
+          />
+      );
+    });
+  }
+
+  _onTimeslotClick(index) {
```

Comment:
This doesn't make any sense, why would I want the index to go up? 

From `Timeslot`:
 -> onTimeslotClick is called with `index`

From `Day`: 
-> takes the index sent by `Timeslot` and looks into it's array. Generates a DateTime string which is passes to `Week`. e.g: `Jun 14th 2015 - 2PM`

From Week:
-> Pass that info to month - doesn't need to touch it

From Month:
-> Pass that info to Calendar - Which adds it to the state.